Added @fosterseth 's remote to the repo. You tripped me up naming it sockceptor! With this branch:

```
tests/api/inventories/test_smart_inventory.py::TestSmartInventory::test_source_inventory_variables_ignored PASSED [100%]
```

with `release_1.1`

```
tests/api/inventories/test_smart_inventory.py::TestSmartInventory::test_source_inventory_variables_ignored FAILED [100%]

============================================= FAILURES =============================================
____________________ TestSmartInventory.test_source_inventory_variables_ignored ____________________
Traceback (most recent call last):
  File "/home/alancoding/repos/tower-qa/tests/api/inventories/test_smart_inventory.py", line 390, in test_source_inventory_variables_ignored
    assert job.status == 'failed'
AssertionError: assert 'error' == 'failed'
  - failed
  + error
-----------
```

It's pretty well beyond dispute that the removed code here was closing the socket, which lead to the empty reads.

--- a/pkg/workceptor/workceptor.go
+++ b/pkg/workceptor/workceptor.go
@@ -524,9 +524,6 @@ func (w *Workceptor) GetResults(ctx context.Context, unitID string, startPos int
 				return
 			}
 			for {
-				if unit.Status().State == WorkStateFailed {
-					return
-				}
 				select {
 				case <-ctx.Done():
 					return

